front/app/components/SignUpIn/AuthProviderButton.tsx

```diff
@@ -111,10 +111,20 @@ export interface Props {
   className?: string;
   onContinue: TOnContinueFunction;
   children: React.ReactNode;
+  showConsentOnFlows?: Array<TSignUpInFlow>;
 }
 
```

Comment:
Think it would be nice to decouple the Vienna logic from this Button- now it's creating a lot of dependencies between the two and adding complexity to the core product that should probably be in the module. 

Maybe you could create a new `AuthProviderButton` inside of the module, and put all the Vienna-specific logic in there? And put all the rendered JSX (and other stuff that can be shared between the two) in a subcomponent that can then be reused inside of the module? (if that makes sense)

Reply:
That would also make it easier to add some basic sanity tests for both buttons

Reply:
I was debating about this too, but then how would I deal with all the existing behaviour & styling that the AuthProviderButton already provides, such as tracking SSO logins, the whole Consent Policy etc. Would you duplicate that over?